### PR TITLE
pythonPackages.{pure-python-adb,adb-shell,androidtv}: init

### PR DESCRIPTION
**Pull request by JamieMagee**
###### Motivation for this change

Support `androidtv` in Home Assistant

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/adb-shell/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  src = fetchPypi {
+    inherit version;
+    pname = "adb_shell";
+    sha256 = "0bb9589jv4h38mv3rhxxpmnyc4xvm29brfx3gzgnn51w3wdyc0qc";
+  };
+
+  propagatedBuildInputs = [ aiofiles cryptography libusb1 pyasn1 rsa ];
+
+  # Tests aren't included in pypi package
+  doCheck = false;
```
Comment (mweinelt) [inline pkgs/development/python-modules/adb-shell/default.nix:17]: Then please use fetchFromGitHub instead.

File: pkgs/development/python-modules/adb-shell/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+  meta = with lib; {
+    description =
+      "A Python implementation of ADB with shell and FileSync functionality.";
+    homepage = "https://github.com/JeffLIrion/adb_shell";
+    license = licenses.asl20;
```
Comment (mweinelt) [inline pkgs/development/python-modules/adb-shell/default.nix:23]: Add yourself as maintainer.

File: pkgs/servers/home-assistant/component-packages.nix [hunk 3]
```diff
@@ -29,7 +29,7 @@
     "amcrest" = ps: with ps; [ ha-ffmpeg ]; # missing inputs: amcrest
     "ampio" = ps: with ps; [ ]; # missing inputs: asmog
     "android_ip_webcam" = ps: with ps; [ ]; # missing inputs: pydroid-ipcam
-    "androidtv" = ps: with ps; [ ]; # missing inputs: adb-shell[async] androidtv[async] pure-python-adb
+    "androidtv" = ps: with ps; [ adb-shell ]; # missing inputs: androidtv[async] pure-python-adb
```
Comment (mweinelt) [inline pkgs/servers/home-assistant/component-packages.nix:32]: Looks like this is only one of the three libraries to make that component work. You'll also need to package `androidtv` and `pure-python-adb`.

Also the regeneration of the component-packages should be in their own commit, since it touches the home-assistant attribute.

**Review by mweinelt — CHANGES_REQUESTED** [review]
Off to a good start, but does not look complete yet.

File: pkgs/development/python-modules/adb-shell/default.nix (hunk 2 above)
Comment (JamieMagee) [reply pkgs/development/python-modules/adb-shell/default.nix:23]: Should that come as a separate PR, or as part of this PR?

File: pkgs/servers/home-assistant/component-packages.nix (hunk 3 above)
Comment (JamieMagee) [reply pkgs/servers/home-assistant/component-packages.nix:32]: Okay, I'll exclude this change from the PR. I also plan on packaging the other 2 libraries as well.

File: pkgs/development/python-modules/adb-shell/default.nix (hunk 2 above)
Comment (mweinelt) [reply pkgs/development/python-modules/adb-shell/default.nix:23]: If you are not already listed as a maintainer, add yourself in a separate commit. Then reference yourself in the commit that adds the new package.

Comment (JamieMagee) [reply pkgs/development/python-modules/adb-shell/default.nix:23]: @mweinelt Can you please review #98478 as well?

Comment (mweinelt) [reply pkgs/development/python-modules/adb-shell/default.nix:27]: Done, please rebase this branch.

File: pkgs/servers/home-assistant/component-packages.nix (hunk 3 above)
Comment (mweinelt) [reply pkgs/servers/home-assistant/component-packages.nix]: It can be included in this PR, just not in the same commit.

**Comment by mweinelt** [comment]
We're still missing androidtv and pure-python-adb. The rest looks good.

**Comment by JamieMagee** [comment]
> We're still missing androidtv and pure-python-adb. The rest looks good.

I was planning on opening separate PRs for those. Is that okay?

**Comment by mweinelt** [comment]
Just include them here please, so we can have one pr including the component package regeneration.

**Comment by JamieMagee** [comment]
I've added all the required packages.

File: pkgs/development/python-modules/adb-shell/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{ aiofiles, buildPythonPackage, cryptography, fetchFromGitHub, isPy3k, lib
+, libusb1, mock, pyasn1, python, pycryptodome, rsa }:
+
+buildPythonPackage rec {
+  pname = "adb-shell";
+  version = "0.2.2";
+
```
Comment (mweinelt) [inline pkgs/development/python-modules/adb-shell/default.nix:7]: The tests fail on Python 2.7, so let's disable it for that version.

```
builder for '/nix/store/1fshwl0k0yz59rbwvdj0rlk0hkz6mx38-python2.7-adb-shell-0.2.2.drv' failed with exit code 1; last 10 log lines:
      async def test_adb_connection_error(self):
          ^
  SyntaxError: invalid syntax
  
  
  ----------------------------------------------------------------------
  Ran 66 tests in 4.088s
  
  FAILED (errors=2)
  builder for '/nix/store/1fshwl0k0yz59rbwvdj0rlk0hkz6mx38-python2.7-adb-shell-0.2.2.drv' failed with exit code 1
cannot build derivation '/nix/store/mjqwh7j0h2qin8f4qhwgivwwkabz0iqg-env.drv': 1 dependencies couldn't be built
[8 built (1 failed), 45 copied (96.3 MiB), 23.6 MiB DL]
error: build of '/nix/store/mjqwh7j0h2qin8f4qhwgivwwkabz0iqg-env.drv' failed
```

**Review by mweinelt — CHANGES_REQUESTED** [review]
The packages look pretty good, please disable `adb-shell` on py27 and regenerate the component packages for home-assistant.

**Comment by mweinelt** [comment]
> I'll make the changes to home-assistant in a followup PR. Home assistant needs a version bump anyway as `0.115.x` has been released.

I've taken care of the home-assistant update in #99217. You can regenerate the component-packages now and I'll rebase on top of that, or we do it the other way around. I don't mind.

**Comment by JamieMagee** [comment]
I think I'll just wait until your PR goes in, and then I'll rebase on top of that. I'm in no rush to get this merged.

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 98452` [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>python37Packages.adb-shell</li>
    <li>python37Packages.androidtv</li>
    <li>python37Packages.pure-python-adb</li>
    <li>python38Packages.adb-shell</li>
    <li>python38Packages.androidtv</li>
    <li>python38Packages.pure-python-adb</li>
  </ul>
</details>

**Review by mweinelt — APPROVED** [review]

File: pkgs/development/python-modules/pure-python-adb/default.nix [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "0kdr7w2fhgjpcf1k3l6an9im583iqkr6v8hb4q1zw30nh3bqkk0f";
+  };
+
+  # Disable tests as they require docker, docker-compose and a dedicated
+  # android emulator
+  doCheck = false;
+
```
Comment (mweinelt) [inline pkgs/development/python-modules/pure-python-adb/default.nix:17]: Whoops, we're missing a `pythonImportsCheck` here.

Last gotcha.

**Review by mweinelt — CHANGES_REQUESTED** [review]
Last nit.

**Review by mweinelt — APPROVED** [review]
